gnosis/gp-ui
# Lowercase order id when searching

# Summary

Closes #309 

Order id search is now case insensitive

## Update 2021/03/24

`orderid` is extracted from URL via regex. As long as the orderId is in there continuously, everything around it is ignored.
Example:
- https://pr323--gpui.review.gnosisdev.com/orders/%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%800XC31de83790b0a0cff23bda533ab488fda0029fb7cafffe6d0b0a7d06bf957f0b9367280befb2464064d4db7ffa6440ede651df48604beb05%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80%F0%9F%9A%80/
- https://pr323--gpui.review.gnosisdev.com/rinkeby/orders/kjldfasldhflasdhfa0xc31de83790b0a0cff23bda533ab488fda0029fb7cafffe6d0b0a7d06bf957f0b9367280befb2464064d4db7ffa6440ede651df48604beb0589273n4c12b3b1f19234/
- https://pr323--gpui.review.gnosisdev.com/rinkeby/orders/"0xc31de83790b0a0cff23bda533ab488fda0029fb7cafffe6d0b0a7d06bf957f0b9367280befb2464064d4db7ffa6440ede651df48604beb05"/

Same applies to search input. Anything in there will be added to the URL, which in turn will be "cleaned" before passing it down to the query.
In case there's no match, it'll fail, as expected.

# Testing

1. Open the app
2. Copy any existing order id, such as `0xc31de83790b0a0cff23bda533ab488fda0029fb7cafffe6d0b0a7d06bf957f0b9367280befb2464064d4db7ffa6440ede651df48604beb05`
3. Make any char uppercase
4. Paste it into the search bar and search

- [ ] The order should load

5. On the order details page URL, make any char uppercase:
![screenshot_2021-03-23_13-47-09](https://user-images.githubusercontent.com/43217/112215989-3ef24f00-8bde-11eb-9a9d-eb1d9e5e0c01.png)
6. Hit enter

- [ ] The same order should load
